src/fsharp/PatternMatchCompilation.fs

```diff
@@ -154,6 +155,24 @@ let rec pathEq p1 p2 =
     | PathEmpty(_), PathEmpty(_) -> true
     | _ -> false
 
+//// (Temporarily copy-pasted from TypeChecker.fs)
```

Comment:
(just an opinion) You can expose it for use in TypeChecker.fs, and replace the comment akin to "Comes from TypeChecker.fs, might move to more apropriate place later.".

This would be better than having the code duplicated, and unless it becomes used in a third place, the location is not too important.

Reply:
@smoothdeveloper I kind of like that idea actually. That way its pretty obvious that not much code depends on it.